src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -76,6 +76,10 @@ public interface ProgrammingExerciseRepository extends JpaRepository<Programming
             "auxiliaryRepositories" })
     Optional<ProgrammingExercise> findWithTemplateAndSolutionParticipationSubmissionsAndResultsAndAuxiliaryRepositoriesById(Long exerciseId);
 
+    @EntityGraph(type = LOAD, attributePaths = { "categories", "teamAssignmentConfig", "templateParticipation.submissions.results", "solutionParticipation.submissions.results",
+            "auxiliaryRepositories", "plagiarismDetectionConfig", "buildPlanConfiguration", "buildScript", "templateParticipation", "solutionParticipation" })
+    Optional<ProgrammingExercise> findForCreationById(Long exerciseId);
+
     @EntityGraph(type = LOAD, attributePaths = "testCases")
     Optional<ProgrammingExercise> findWithTestCasesById(Long exerciseId);
```

Comment:
```suggestion
    Optional<ProgrammingExercise> findWithTestCasesById(long exerciseId);
```

Reply:
There are quite many instances in this file, where `Long` is used. I am not sure, if there is a reason why `Long` or `long` are used in those cases.
Nevertheless, unifying the data types used in the code is probably out of scope of this PR and should be done separately.